packages/vscode-extension/src/core/sdksetup/SdkServices.ts

```diff
@@ -24,9 +24,33 @@ const translationService = new VSTranslationService();
 const fileSystemService = new FileSystemService();
 
 export async function installIfNeeded() {
-	if (!fs.existsSync(path.join(sdkPath)) || !(await validateSdk(path.join(sdkPath)))) {
-		await install();
-	}
+    validateJavaVersion();
+
+    if (!fs.existsSync(path.join(sdkPath)) || !(await validateSdk(path.join(sdkPath)))) {
+        await install();
+    }
+}
+
+function validateJavaVersion() {
+    const environmentInformationService = new EnvironmentInformationService();
+    const installedJavaVersion = environmentInformationService.getInstalledJavaVersionString();
+    const requiredJavaVersion = `${ApplicationConstants.SDK_REQUIRED_JAVA_VERSION}`;
+    if (installedJavaVersion.startsWith(requiredJavaVersion)) {
+        return;
+    }
+
+    let messageKey;
+    let errorMessage;
+    if (installedJavaVersion === '') {
+        messageKey = `${ERRORS.SDK_JAVA_VERSION_NOT_INSTALLED}`;
+        errorMessage = translationService.getMessage(messageKey, requiredJavaVersion);

```

Comment:
We can remove the messageKey variable and directly use `errorMessage = translationService.getMessage(ERRORS.SDK_JAVA_VERSION_NOT_INSTALLED, requiredJavaVersion);`  here and `    errorMessage = translationService.getMessage(ERRORS.SDK_JAVA_VERSION_NOT_COMPATIBLE, installedJavaVersion, requiredJavaVersion);` below in line 51.

Reply:
Updated.